packages/flutter/lib/src/widgets/image.dart

```diff
@@ -768,8 +768,8 @@ class Image extends StatefulWidget {
   ///     ),
   ///     child: Image.network(
   ///       'https://flutter.github.io/assets-for-api-docs/assets/widgets/puffin.jpg',
-  ///       frameBuilder: (BuildContext context, Widget child, int? frame, bool? wasSynchronouslyLoaded) {
-  ///         if (wasSynchronouslyLoaded ?? false) {
+  ///       frameBuilder: (BuildContext context, Widget child, int? frame, bool wasSynchronouslyLoaded) {
```

Comment:
/cc @a14n could there (or maybe is there) a lint that would have caught this?

Reply:
Do you mean: a lint checking that types used in lambda should have exactly the same type as the expected types ?

I'm not aware of such a lint (mostly because outside Flutter [Effective Dart recommand to not annotate inferred parameter types on function expressions](https://dart.dev/guides/language/effective-dart/design#dont-annotate-inferred-parameter-types-on-function-expressions)). I will take a look to craft one.

Reply:
#78534